File: golang/01-magic-receipt/receipt.go
package kata

import "math"

type Receipt struct{}

func NewReceipt() *Receipt {
	return &Receipt{}
}

func (r *Receipt) calculate_total(items []float64, customerType string) float64 {
	total := 0.0
	for i := 0; i < len(items); i++ {
		total += items[i]
	}
	discount := 0.0
	if customerType == "member" {
		discount = total * 0.05
	} else if customerType == "vip" {
		discount = total * 0.15
	}
	if total > 50 {
		discount += 5
	}
	final := (total - discount) * 1.08
	if customerType == "vip" {
		final -= 2
	}
	return math.Round(final*100) / 100
}

File: csharp/01-magic-receipt/src/receipt.cs
public class Receipt
{
    public decimal calculate_total(decimal[] items, string customerType = "")
    {
        decimal total = 0;
        for (int i = 0; i < items.Length; i++)
        {
            total += items[i];
        }

        decimal discount = 0;
        if (customerType == "member")
        {
            discount = total * 0.05m;
        }
        else if (customerType == "vip")
        {
            discount = total * 0.15m;
        }

        if (total > 50)
        {
            discount += 5;
        }

        var final = (total - discount) * 1.08m;
        if (customerType == "vip")
        {
            final -= 2;
        }

        return Math.Round(final, 2);
    }
}
